Assets/Scripts/UI/PickupPup/DogProfile/DogAdoptProfile.cs

```diff
@@ -54,6 +61,14 @@ public override void SetProfile(Dog dog)
 
     #endregion
 
+    void handleAdoptEvent(string eventName, Dog dog)
+    {
+        if (eventName == k.ADOPT && dog.Info == dogInfo)
```

Comment:
May want to use `dog.Info.Equals()` instead so we don't run into any instance copy errors